app/src/app/framework/category/category.component.ts

```diff
@@ -62,76 +62,136 @@ export class CategoryComponent implements OnInit, AfterViewInit, OnDestroy {
+      data: 'label',
+      cssClasses: ['text-center'],
+      searchable: false,
+      sortable: true
+    } as unknown as Column;
+
+    const parentColumn = {
+      display: 'Parent ID',
+      name: 'parent',
+      data: 'parent',
+      cssClasses: ['text-center'],
+      style: { width: '6rem' },
+      searchable: false,
+      sortable: true
+    } as unknown as Column;
+
+    this.tableColumns = [publicColumn, labelColumn, nameColumn, parentColumn];
+  }
+
+  /**
+   * Load categories from the backend.
+   */
+  private loadCategories(): void {
+    this.categoryService.getCategoryIteration(this.apiParameters).pipe(
+      takeUntil(this.unSubscribe)).subscribe((response: APIGetMultiResponse<CmdbCategory>) => {
+      this.categoriesAPIResponse = response;
+      this.categories = this.categoriesAPIResponse.results;
+      this.totalResults = response.total;
+    });
+  }
+
+  /**
+   * Load category tree from the backend.
+   */
+  private loadCategoryTree(): void {
+    this.displayMode.subscribe(() => {
+      this.sidebarService.categoryTree.asObservable().pipe(takeUntil(this.unSubscribe))
+        .subscribe((categoryTree: CmdbCategoryTree) => {
+          this.categoryTree = categoryTree;
       });
     });
-    this.dtOptions = {
-      columns: [
-        {
-          title: 'PublicID',
-          name: 'public_id',
-          data: 'public_id'
-        },
-        {
-          title: 'Name',
-          name: 'name',
-          data: 'name'
-        },
-        {
-          title: 'Label',
-          name: 'label',
-          data: 'label'
-        },
-        {
-          title: 'ParentID',
-          name: 'parent',
-          data: 'parent'
-        }
-      ],
-      searching: false,
-      serverSide: true,
-      processing: true,
-      ajax: (params: any, callback) => {
-
-        const apiParameters: CollectionParameters = {
-          page: Math.ceil(params.start / params.length) + 1,
-          limit: params.length,
-          sort: params.columns[params.order[0].column].name,
-          order: params.order[0].dir === 'desc' ? -1 : 1,
-        };
-
-        this.categoryService.getCategoryIteration(apiParameters).pipe(
-          takeUntil(this.unSubscribe)).subscribe((response: APIGetMultiResponse<CmdbCategory>) => {
-          this.categoriesAPIResponse = response;
-          this.categories = this.categoriesAPIResponse.results;
-          callback({
-            recordsTotal: response.total,
-            recordsFiltered: response.total,
-            data: this.categories
-          });
-        });
-      }
-    };
   }
 
-  public ngAfterViewInit(): void {
-    this.dtTrigger.next();
+  /**
+   * Will generate all needed data for cmdb-table.
+   */
+  private dataLoader(): void {
+    this.tableColumnBuilder();
+    this.loadCategoryTree();
+    this.loadCategories();
+  }
+
+  public ngOnInit(): void {
+    this.dataLoader();
+  }
+
+  /**
+   * On table sort change.
+   * Reload all objects.
```

Comment:
You still have objects here in the docs. I guess you copied this from the object type lists.